File: models/Trip.js
const mongoose = require('mongoose');
const db = require('./db');

const tripSchema = new mongoose.Schema({
    route_id: 'string',
    trip_id: 'string',
    trip_headsign: 'string',
    direction_id: 'number',
    shape_id: 'number',
    shape_path: [{
        lat: 'number',
        lng: 'number',
        dist: 'number'
    }]
});

const Trip = db.model('trips', tripSchema);

module.exports = Trip;

File: lagom-api/spbus-api/src/main/scala/br/ufc/insightlab/spbus/api/model/Trip.scala
package br.ufc.insightlab.spbus.api.model

import play.api.libs.functional.syntax._
import play.api.libs.json._

case class Trip(directionId: Int,
                routeId: String,
                shapeId: Int,
                shapePath: List[PathPoint],
                tripHeadsign: String,
                tripId: String)

object Trip {
  implicit val formatReads: Reads[Trip] =
    (JsPath \ "direction_id").format[Int]
      .and((JsPath \ "route_id").format[String])
      .and((JsPath \ "shape_id").format[Int])
      .and((JsPath \ "shape_path").format[List[PathPoint]])
      .and((JsPath \ "trip_headsign").format[String])
      .and((JsPath \ "trip_id").format[String])
      .apply(Trip.apply, unlift(Trip.unapply))

  implicit val formatWrites = Json.writes[Trip]
}
